sktime/forecasting/varmax.py

```diff
@@ -0,0 +1,392 @@
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : array_like
+            The observed time-series process :math:`y`, shaped n_obs x k_endog.
+        X : array_like, optional (default=None)
+            Array of exogenous regressors, shaped n_obs x k.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self._forecaster = _VARMAX(
+            endog=y,
+            exog=X,
+            order=self.order,
+            trend=self.trend,
+            error_cov_type=self.error_cov_type,
+            measurement_error=self.measurement_error,
+            enforce_stationarity=self.enforce_stationarity,
+            enforce_invertibility=self.enforce_invertibility,
+            trend_offset=self.trend_offset,
+        )
+        self._fitted_forecaster = self._forecaster.fit(
+            start_params=self.start_params,
+            transformed=self.transformed,
+            includes_fixed=self.includes_fixed,
+            cov_type=self.cov_type,
+            cov_kwds=self.cov_kwds,
+            method=self.method,
+            maxiter=self.maxiter,
+            full_output=self.full_output,
+            disp=self.disp,
+            callback=self.callback,
+            return_params=self.return_params,
+            optim_score=self.optim_score,
+            optim_complex_step=self.optim_complex_step,
+            optim_hessian=self.optim_hessian,
+            flags=self.flags,
+            low_memory=self.low_memory,
+        )
+        return self
+
+    # defining _predict method to pass in dynamic, information_set and signal_only
+    # which are not passed into the _predict method in _StatsModelsAdapter
+    def _predict(self, fh, X=None):
+        """
+        Wrap Statmodel's VARMAX forecast method.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasters horizon with the steps ahead to to predict.
+            Default is one-step ahead forecast,
+            i.e. np.array([1])
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+
+        Returns
+        -------
+        y_pred : np.ndarray
+            Returns series of predicted values.
+        """
+        start, end = fh.to_absolute_int(self._y.index[0], self.cutoff)[[0, -1]]
+
+        y_pred = self._fitted_forecaster.predict(
+            start=start,
+            end=end,
+            dynamic=self.dynamic,
+            information_set=self.information_set,
+            signal_only=self.signal_only,
+            exog=X,
+        )
+
+        # statsmodel returns zero-based index when index is of type int with the
+        # following warning
+        # ValueWarning: No supported index is available. Prediction results will be
+        # given with an integer index beginning at `start`...
+        # but only when out-of-sample forecasting, i.e. when forecasting horizon is
+        # greater than zero
```

Comment:
I like this thorough explanation! This is very useful for any future people who look at this.